svc/rewards/worker/in_progress-worker.go

```diff
@@ -0,0 +1,105 @@
+package worker
+
+import (
+	"context"
+	"database/sql"
+	"log"
+
+	"github.com/google/uuid"
+
+	lib_solana "github.com/SatorNetwork/sator-api/lib/solana"
+	"github.com/SatorNetwork/sator-api/svc/rewards/interfaces"
+	"github.com/SatorNetwork/sator-api/svc/rewards/repository"
+)
+
+type (
+	InProgressTransactionStatusWorkerJob struct {
+		UserID uuid.UUID
+		Amount float64
+		TxHash string
+	}
+
+	InProgressTransactionStatusWorker struct {
+		transactions chan InProgressTransactionStatusWorkerJob
+
+		ctx  context.Context
+		repo interfaces.RewardsRepository
+		sc   solanaClient
+
+		done chan struct{}
+	}
+
+	solanaClient interface {
+		GetTransaction(ctx context.Context, txhash string) (lib_solana.GetConfirmedTransactionResponse, error)
+	}
+)
+
+func NewInProgressTransactionStatusWorker(
+	ctx context.Context,
+	repo interfaces.RewardsRepository,
+	sc solanaClient,
+) *InProgressTransactionStatusWorker {
+	return &InProgressTransactionStatusWorker{
+		transactions: make(chan InProgressTransactionStatusWorkerJob),
+		ctx:          ctx,
+		repo:         repo,
+		sc:           sc,
+		done:         make(chan struct{}),
+	}
+}
+
+func (w *InProgressTransactionStatusWorker) Start() {
+LOOP:
+	for {
+		select {
+		case transaction := <-w.transactions:
+			resp, err := w.sc.GetTransaction(w.ctx, transaction.TxHash)
+			if err != nil {
+				log.Println(err)
+				continue
+			}
+
+			if !(resp.Meta.Err == nil) {
```

Comment:
I think we can simplify:
`resp.Meta.Err != nil` instead of `!(resp.Meta.Err == nil)`